cypress-io/cypress
# misc: Close extra tabs/windows between tests

### Additional details

Closes any extra targets (tabs/windows) between tests so that new tests always start with only the main Cypress tab, in order to maintain test isolation and a clean slate before each test. This is in anticipation of support for testing multiple tabs/windows with the future `@cypress/puppeteer` plugin.

### Steps to test


- Run a Cypress spec in open mode that has at least a couple tests in it
- Open a new browser tab while the tests are running
- You should see the tab you opened automatically close before the next test starts running

### How has the user experience changed?


### PR Tasks


- [N/A] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [N/A] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
